docs/docs/guides/web3_providers_guide/index.md

```diff
@@ -0,0 +1,173 @@
+2. For contracts `web3.Contract.setProvider` can be used to set provider for **all instances** of contracts created by `web3.eth.Contract`.
+
+---
+
+## Examples
+
+### Local Geth Node
+
+```ts
+const Web3 = require('web3');
+const web3 = new Web3('http://localhost:8545');
+// or
+const web3 = new Web3(new Web3.providers.HttpProvider('http://localhost:8545'));
+
+// change provider
+web3.setProvider('ws://localhost:8546');
+// or
+web3.setProvider(new Web3.providers.WebsocketProvider('ws://localhost:8546'));
+
+// Using the IPC provider in node.js
+const net = require('net');
+const web3 = new Web3('/Users/myuser/Library/Ethereum/geth.ipc', net); // mac os path
+// or
+const web3 = new Web3(
+	new Web3.providers.IpcProvider('/Users/myuser/Library/Ethereum/geth.ipc', net),
+); // mac os path
+// on windows the path is: "\\\\.\\pipe\\geth.ipc"
+// on linux the path is: "/users/myuser/.ethereum/geth.ipc"
+```
+
+### Remote Node Provider
+
+```ts
+// Using a remote node provider, like Alchemy (https://www.alchemyapi.io/supernode), is simple.
+const Web3 = require('web3');
+const web3 = new Web3('https://eth-mainnet.alchemyapi.io/v2/your-api-key');
+```
+
+### Injected providers
+
+As stated above, the injected provider should be in compliance with [EIP-1193](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1193.md). And it is tested with Ganache provider, Hardhat provider, and Incubed (IN3) as a provider.
+
+The web3.js 4.x Provider specifications are defined in [web3 base provider](https://github.com/ChainSafe/web3.js/blob/4.x/packages/web3-types/src/web3_base_provider.ts) for Injected Providers.
+
+```ts
+const Web3 = require('web3');
+// Using an EIP1193 provider like MetaMask can be injected
+
+if (window.ethereum) {
+	// Check if ethereum object exists
+	await window.ethereum.request();
+	window.web3 = new Web3(window.ethereum); // inject provider
+}
+```
+
+### Provider Options
+
+There are differences in the objects that could be passed in the Provider constructors.
+
+#### HttpProvider
+
+The options is of type `HttpProviderOptions`, which is an object with a single key named `providerOptions` and its value is an object of type `RequestInit`.
+Regarding `RequestInit` see [microsoft's github](https://microsoft.github.io/PowerBI-JavaScript/interfaces/_node_modules_typedoc_node_modules_typescript_lib_lib_dom_d_.requestinit.html).
+
+For example:
+
+```ts
+const httpOptions = {
+	providerOptions: {
+		body: undefined,
+		cache: 'force-cache',
+		credentials: 'same-origin',
+		headers: {
+			'Content-Type': 'application/json',
+		},
+		integrity: 'foo',
+		keepalive: true,
+		method: 'GET',
+		mode: 'same-origin',
+		redirect: 'error',
+		referrer: 'foo',
+		referrerPolicy: 'same-origin',
+		signal: undefined,
+		window: undefined,
+	} as RequestInit,
+};
+```
+
+#### WebSocketProvider
```

Comment:
The same thing for IpcProvider. So we can change the title to `WebSocketProvider and IpcProvider`

Reply:
Actually, there are some differences that will be addressed as part of #5737